src/main/java/com/google/finapp/FinAppService.java

```diff
@@ -33,24 +32,28 @@
   }
 
   @Override
-  public void createCustomer(Customer customer, StreamObserver<Empty> responseObserver) {
+  public void createCustomer(Customer customer, StreamObserver<CustomerResponse> responseObserver) {
+    ByteArray customerId;
     try {
-      spannerDao.createCustomer(
+      customerId = spannerDao.createCustomer(
           UuidConverter.getBytesFromUuid(UUID.randomUUID()),
           customer.getName(),
           customer.getAddress());
     } catch (SpannerDaoException e) {
       responseObserver.onError(Status.fromThrowable(e).asException());
       return;
     }
-    responseObserver.onNext(Empty.getDefaultInstance());
+    CustomerResponse response = CustomerResponse.newBuilder()
+        .setId(ByteString.copyFrom(customerId.toByteArray())).build();
+    responseObserver.onNext(response);
     responseObserver.onCompleted();
   }
 
   @Override
-  public void createAccount(Account account, StreamObserver<Empty> responseObserver) {
+  public void createAccount(Account account, StreamObserver<AccountResponse> responseObserver) {
+    ByteArray accountId;
```

Comment:
Here, you could assign `accountId` to the UUID and then remove the assignment below. Then, make the SpannerDao `createAccount`s return nothing.